thunder/extend/__init__.py

```diff
@@ -264,8 +281,44 @@ def register_implementation(
             symbol=op, checker=checker, execution_transform=execution_transform, grad_transform=grad_transform
         )
 
-        id = sym_or_id.id if isinstance(sym_or_id, Symbol) else sym_or_id
-        self.implmap[id] = impl
+        _id = sym_or_id.id if isinstance(sym_or_id, Symbol) else sym_or_id
+        self.implmap[_id] = impl
+
+
+class SingleOpExecutor(OperatorExecutor):
```

Comment:
TBH, I'd prefer this to be a factory function rather than a subclass. It is one thing to have a convenience function (which will only affect the caller) and another to have a new datatype (which might not have grave repercussions, but has a much wider impact radius).

Reply:
Yeah, that makes sense to me. Come to think of it I largely feel the same. Object hierarchies are a path to pain.